### ISSUE TITLE
indent rule applied inside of multi-line `template` strings

### ISSUE THREAD

**Issue by nonplus** [palantir/tslint#1611]
### Bug Report
- **TSLint version**: `3.15.1`
- **TypeScript version**: `2.0.3`
- **Running TSLint via**: CLI
#### TypeScript code being linted

``` ts
// (second line in the template string has 3 leading spaces)
let sql = `SELECT *
   FROM user
   WHERE name = 'bob'`;
```

with `tslint.json` configuration:

``` json
{
    "rules": {
        "indent": [
            true,
            "tabs"
        ]
}
```
#### Actual behavior

TSLint logs an `(indent) db/test.ts[1, 1]: tab indentation expected` error.
#### Expected behavior

No error should be logged.
#### Workaround:

``` ts
/* tslint:disable:indent */
let sql = `SELECT *
   FROM user
   WHERE name = 'bob'`;
/* tslint:enable:indent */
```
#### Notes:

Indentation should be ignored inside the literal portion of a template string (`InTemplateHeadOrNoSubstitutionTemplate` and `InTemplateMiddleOrTail`).

However, I think it makes sense to check for indentation **within template substitution**.  So the following should be considered an indentation error:

``` ts
let name = "Bob";
// (second line in the template string has 3 leading spaces)
let s = `Hello ${
   name
}, how are you?`;
```

**Comment by JoshuaKGoldberg** [comment]
Looks like this was fixed! 🎉
